src/test/java/picard/sam/SamErrorMetric/CollectSamErrorMetricsTest.java

```diff
@@ -128,15 +140,52 @@ private static void parseDirective0(final String directive, final String extensi
         }
     }
 
-    @DataProvider(name = "OneCovariateErrorMetricsDataProvider")
-    public Object[][] oneCovariateErrorMetricsDataProvider() {
-        final File simpleSamWithBaseErrors1 = new File(TEST_DIR, "simpleSamWithBaseErrors1.sam");
-        final File simpleSamWithBaseErrors2 = new File(TEST_DIR, "simpleSamWithBaseErrors2.sam");
-        final File simpleSingleStrandConsensusSamWithBaseErrors = new File(TEST_DIR, "simpleSingleStrandConsensusSamWithBaseErrors.sam");
-        final File simpleDuplexConsensusSamWithBaseErrors = new File(TEST_DIR, "simpleDuplexConsensusSamWithBaseErrors.sam");
-        final File chrMReadsWithClips = new File(TEST_DIR, "chrMReadsWithClips.sam");
-        final int priorQ = 30;
+    private final Map<File, File> errorMetrics = new HashMap<>();
+
+    private final File simpleSamWithBaseErrors1 = new File(TEST_DIR, "simpleSamWithBaseErrors1.sam");
+    private final File simpleSamWithBaseErrors2 = new File(TEST_DIR, "simpleSamWithBaseErrors2.sam");
+    private final File simpleSingleStrandConsensusSamWithBaseErrors = new File(TEST_DIR, "simpleSingleStrandConsensusSamWithBaseErrors.sam");
+    private final File simpleDuplexConsensusSamWithBaseErrors = new File(TEST_DIR, "simpleDuplexConsensusSamWithBaseErrors.sam");
+    private final File chrMReadsWithClips = new File(TEST_DIR, "chrMReadsWithClips.sam");
+
+    @BeforeTest()
+    public void samMetricsProvider() {
+        final File[] files = new File[]{
+                simpleSamWithBaseErrors1,
+                simpleSamWithBaseErrors2,
+                simpleSingleStrandConsensusSamWithBaseErrors,
+                simpleDuplexConsensusSamWithBaseErrors,
+                chrMReadsWithClips};
+
+        OUTPUT_DATA_PATH = IOUtil.createTempDir("CollectSamErrorMetricsTest", null);
```

Comment:
This is run before every test, so this will create a new temp dir for each test, and store it in a static field. Is that what you want to do here? If each test gets its own temp dir, using a non-static field would make that clearer.